src/dom/index.js

```diff
@@ -15,8 +19,9 @@ import options from '../options';
  * Properties Preact adds to elements it creates
  * @typedef PreactElementExtensions
  * @property {string} [normalizedNodeName] A normalized node name to use in diffing
+ * @property {string} [splitText]
```

Comment:
`splitText` is a property of `Text` nodes: https://developer.mozilla.org/en-US/docs/Web/API/Text/splitText

Reply:
I'm aware of that. The problem is that in the cases where `splitText` is getting used, it's on data that sometimes might be an object of some type, or a number, or a string. That's fine for JavaScript because it will try to coerce the data. Take this example:

```javascript
export function isSameNodeType(node, vnode, hydrating) {
  if (typeof vnode==='string' || typeof vnode==='number') {
    return node.splitText !== undefined;
  }
  if (typeof vnode.nodeName==='string') {
    return !node._componentConstructor && isNamedNode(node, vnode.nodeName);
  }
  return hydrating || node._componentConstructor===vnode.nodeName;
}
```
The above function's first argument is a node. This could be of type 1 or 3, or just a number. A type checker will look at the use of `splitText` and think, `node might be a number or might be a string. I don't know.` Because type checking happens during code or build time, the checker can't tell what the actual value of `node` is. It knows you can't call `splitText` on a number or a DOM node, so it will complain about the use not being valid for them. In contrast, during runtime in the browser, when you call `splitText` on a number and the browser goes, `So, this is a number, but you what to use a string function. Let me see if it has a `toString` method. It does. I'll convert it to a string.` And everything is fine.

The only way to prevent these types of errors is to either escape code with [""] or do a conversion of number to string with something like:

```javascript
if (typeof vnode==='string' || typeof vnode==='number') {
  // Manually convert numbers to string:
  if (typeof node === 'number') node = node.toString()
  return node.splitText !== undefined;
}
``` 
This makes type checkers happy because there is no question that the value using `splitText` is indeed a string. Hope that's clear. Making JavaScript code type safe can be a PITA because its forgiving nature encourages letting the browser handle coercion for us.